app/js/RequestLogger.js

```diff
@@ -1,83 +1,90 @@
 const logger = require('logger-sharelatex')
 const metrics = require('metrics-sharelatex')
 
-module.exports = {
-  errorHandler,
-  middleware
-}
-
-function errorHandler(err, req, res, next) {
-  req._logInfo.set('error', err)
-  res
-    .send(err.message)
-    .status(500)
-    .end()
-}
+class RequestLogger {
+  constructor() {
+    this.errorHandler = this.errorHandler.bind(this)
+    this.middleware = this.middleware.bind(this)
+    this._logInfo = {}
+    this._logMessage = 'http request'
+  }
 
-function middleware(req, res, next) {
-  const startTime = new Date()
+  attach(app) {
+    app.use(this.middleware)
+    app.use(this.errorHandler)
+  }
 
-  // methods to allow the setting of additional information to be logged for the request
-  req._logInfo = {}
-  req._logMessage = 'http request'
-  req.addLogField = function(field, value) {
-    req._logInfo[field] = value
+  errorHandler(err, req, res, next) {
+    this._logInfo.error = err
+    res
+      .send(err.message)
+      .status(500)
+      .end()
   }
-  req.addLogFields = function(fields) {
-    Object.assign(req._logInfo, fields)
+
+  addFields(fields) {
+    Object.assign(this._logInfo, fields)
   }
-  req.setLogMessage = function(message) {
-    req._logMessage = message
+
+  setMessage(message) {
+    this._logMessage = message
   }
 
-  // override the 'end' method to log and record metrics
-  const end = res.end
-  res.end = function() {
-    // apply the standard request 'end' method before logging and metrics
-    end.apply(this, arguments)
+  middleware(req, res, next) {
+    const startTime = new Date()
+    req.requestLogger = this
```

Comment:
Neat idea to avoid creating a separate object for each request, but this won't work with concurrent requests, right? They'll all race to modify the same _logMessage and _logInfo?

Reply:
You are correct, this code is nonsense. I'll change it. :+1: